iceoryx_posh/test/moduletests/test_capro_message.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/internal/capro/capro_message.hpp"
+#include "iceoryx_posh/internal/roudi/introspection/port_introspection.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::capro;
+
+class CaproMessage_test : public Test
+{
+  public:
+    CaproMessage_test()
+    {
+    }
+
+    virtual ~CaproMessage_test()
+    {
+    }
+
+    virtual void SetUp(){};
+    virtual void TearDown(){};
+};
+
+
+TEST_F(CaproMessage_test, CTorSetsParametersCorrectly)
+{
+    uint16_t testServiceID = 1;
+    uint16_t testEventID = 2;
+    uint16_t testInstanceID = 3;
+    ServiceDescription sd = ServiceDescription(testServiceID, testEventID, testInstanceID);
+    iox::popo::ReceiverPortData recData;
+
+    CaproMessage testObj = CaproMessage(CaproMessageType::OFFER, sd, CaproMessageSubType::SERVICE, &recData);
+
+    EXPECT_EQ(&recData, testObj.m_requestPort);
+    EXPECT_EQ(CaproMessageType::OFFER, testObj.m_type);
+    EXPECT_EQ(CaproMessageSubType::SERVICE, testObj.m_subType);
+    EXPECT_EQ(nullptr, testObj.m_chunkQueueData);
+    EXPECT_EQ(0u, testObj.m_historyCapacity);
+    EXPECT_EQ(sd, testObj.m_serviceDescription);
+}
+
+
+TEST_F(CaproMessage_test, DefaultCtorArgs)
```

Comment:
I do not fully understand the test. We do not call a default Ctor here but we call it with a default ServiceDescription. Maybe the test needs a better name, such as ```constructWithDefaultServiceDescription```. Also, does it make sense to check more fields in the via EXPECT afterwards? (I am not sure here, we only need to check what is specific to this case and was not tested elsewhere.)

Reply:
This test is to check default arguments in constructor if initialized properly, 
ServiceDescription is not default argument here, so it is passed(default/initialized may not matter), 
and we are checking for other two (CaproMessageSubTyp , ReceiverPortData) if  initialized properly.
for name ConstructWithDefaultArguments is fine?

Reply:
CaproMessageType is also passed as it is not a default argument

Reply:
Yes, but the main point seems to be that we initialize it with a default constructed ServiceDescription, i.e. ```ServiceDescription{}```. I was misled to believe it has something to do with the default constructor. I would just change the name of the test (e.g. as suggested) to make that more clear.